### ISSUE TITLE
Is there a singularity container available?

### ISSUE THREAD

**Issue by changchuanjun** [jkimlab/PAPipe#2]
Thank you for this nice software, but I can't use it without sudo permission. After trying for a long time, I still can't solve this problem, so I would like to ask if you have a singularity container that does not require root permission that can be used. Looking forward to your reply, thank you

**Comment by jackslinde35** [comment]
We are running into similar issues. An install without docker would be amazing!

**Comment by nayoung9** [comment]
Hi, Thank you for your interest in PAPipe, and sorry for the delayed response. Originally, we did not support Singularity containers. However, I've built a Singularity image using the current PAPipe Docker image. Please try using the provided Singularity image and let us know if you encounter any issues.

Here's how you can get started:

1. Download the Singularity Image File (SIF)
`wget http://bioinfo.konkuk.ac.kr/PAPipe/PAPipe.sif`
2. Run the container:
`singularity shell --writable-tmpfs PAPipe.sif`

Please note that not all functionalities have been thoroughly tested, and this image is provided for temporary use. We plan to add official support in a future update.

If you have any problems using the PAPipe, feel free to let us know. We appreciate your feedback and support.

Thank you.

@changchuanjun @jackslinde35

**Comment by changchuanjun** [comment]
@nayoung9 ，Hi, thank you for providing this temporary singularity for utilizing its functions. Moreover, I am very excited to hear that you will launch an available singularity soon.

**Comment by jackslinde35** [comment]
@nayoung9 - Thanks for your efforts! We have been troubleshooting and close to getting it to run on our cluster. This pipeline is powerful and we look forward to getting it figured out.

**Comment by jackslinde35** [comment]
@nayoung9 @changchuanjun - Getting it to work on a HPC cluster is extremely difficult and not straightforward, especially without sudo. I wonder if there is an install without containers is in the works? That would be amazing. We have exhausted many different methods to get it running and have so far failed. If we find success, I will post here any progress we make.

**Comment by changchuanjun** [comment]
@jackslinde35 Yes, I agree with you. In general, a container consists of a series of software components and their dependencies. Perhaps we could extract them from the container to local termination and utilize them locally.

**Comment by changchuanjun** [comment]
@jackslinde35 Yes, I agree with you. In general, a container consists of a series of software components and their dependencies. Perhaps we could extract them from the container to local termination and utilize them locally.

**Comment by nayoung9** [comment]
I understand that if Singularity is installed on your cluster, you can download images and run containers without needing sudo privileges. When you use the Singularity shell command to access a container, shouldn't the necessary pipelines and programs already be set up within it? 
But I'm not very familiar with Singularity or your specific environment, so I might be missing something.

**Comment by changchuanjun** [comment]
@nayoung9 , yes, you are right👍
